endaq/device/schemata/flash_package.xml

```diff
@@ -28,18 +28,27 @@
             </MasterElement>
         </MasterElement>
 </MasterElement>
-	
-	<!-- Only the following elements must exist in the update package -->
+
+		<!-- Only the following elements must exist in the update package -->
 <MasterElement name="UpdatePkg" id="0xF0" mandatory="1" multiple="0">
     <UIntegerElement name="MinHWRev" id="0xF1" multiple="0" mandatory="1"> Minimum Hardware Rev this update can be applied to</UIntegerElement>
     <UIntegerElement name="FWRev" id="0xF2" multiple="0" mandatory="1"> FW Revision contained in this package (may be 0 for non firmware updates)</UIntegerElement>
     <UIntegerElement name="MinFWRev" id="0xF3" multiple="0" mandatory="0"> Minimum firmware version that can receive this update (in case of interface changes)</UIntegerElement>
 	<IntegerElement name="KeySlot" id="0xF4" multiple="0" mandatory="1"> AES Key slot used to encrypt payload (-1 for unencrypted payload)</IntegerElement>
 	<IntegerElement name="PayloadLen" id="0xF5" multiple="0" mandatory="1"> Length of following encrypted payload  of the following format: 1 or more of (uint16 byte offset || uint16 length || [length] bytes of binary data to flash)</IntegerElement>
-    <StringElement name="TargetProcessor" id="0xF6" multiple="0" mandatory="1"> Processor the attached update is intended for. Values include "EFM32GG11B820F2048GL120", "STM32U585AII6", or "ANY" for non-firmware updates</StringElement>
+    <StringElement name="McuType" id="0xF6" multiple="0" mandatory="1"> Processor the attached update is intended for. Values include "EFM32GG11B820F2048GL120", "STM32U585AII6", or "ANY" for non-firmware updates</StringElement>
     <StringElement name="FWRevString" id="0xF7" multiple="0" mandatory="0"> Human readable firmware revision</StringElement>
     <StringElement name="UpdateNotes" id="0xF8" multiple="0" mandatory="0"> Human readable notes regarding the update</StringElement>
-    <MasterElement name="NonCriticalElements" global="1" id="0xE0" multiple="0" mandatory="0">The device may skip elements inside NonCriticalElements without stopping the update. Any other unrecognized elements will kill the firmware update.
+    <MasterElement name="NonCriticalElements" global="1" id="0xE0" multiple="0" mandatory="0">The device may skip elements inside NonCriticalElements without stopping the update. Any other unrecognized elements will kill the firmware update.</MasterElement>
+    <!-- Additional element for NCP (EFM currently) information. Not strictly necessary but for newer models allows simultaneous updating of the NCP -->
+    <MasterElement name="NcpUpdate" id="0x80" mandatory="0" multiple="0">
+        <StringElement name="NcpType" id="0x86" multiple="0" mandatory="1"> NCP chip type the update is meant for</StringElement>
+        <StringElement name="Ncp" id="0x89" multiple="0" mandatory="1"> NCP update file</StringElement>
```

Comment:
The element name `<Ncp>` seems odd, since NCP is a prefix to all the other names. It doesn't really hint the element's function.

Reply:
that's a fair point, it should be something like `NCPFile`, but now that we're having an `NCPUpdateData` element, I don't think there's any point to still having this in the schemata anymore.

Reply:
Yeah, this isn't a command, this is the file with the update itself, so a filename argument seems off. Double-check with @pscheidler and/or the firmware engineers, but I agree that it's probably redundant and can be removed.